File: project_euler/Problem076/Problem076.c
#include <stdio.h>
#include <stdlib.h>

void Problem076(void) {
    int target = 100;
    int *ways = (int *) malloc(sizeof(int) * (target + 1));
    for (int i = 1; i <= target; ++i) {
        ways[i] = 0;
    }
    ways[0] = 1;
    for (int i = 1; i < target; ++i) {
        for (int j = i; j <= target; ++j) {
            ways[j] += ways[j - i];
        }
    }
    printf("%d\n", ways[target]);
}

File: project_euler/Problem076/Problem076.js
function Problem076() {
    let target = 100
    let ways = Array(target + 1).fill(0)
    ways[0] = 1
    for (let i = 1; i < target; ++i) {
        for (let j = i; j <= target; ++j) {
            ways[j] += ways[j - i]
        }
    }
    console.log(ways[target])
}

module.exports = {
    Problem076 : Problem076
};
